### PR TITLE
feature(nyp):add R2d2 comments

### PR DESCRIPTION
**Pull request by Will-Nie**
## Description
add some comments for r2d2 to improve readability

## Check List

- [x] merge the latest version source branch/repo, and resolve all the conflicts
- [x] pass style check
- [x] pass all the tests

### CODE REVIEW

File: ding/model/template/q_learning.py [hunk 1]
```diff
@@ -678,11 +678,14 @@ def forward(
         """
 
         x, prev_state = inputs['obs'], inputs['prev_state']
+        # for both inference and other cases, the network structure is encoder -> rnn network -> head
+        # the difference is inference take the data with seq_len=1 (or T = 1)
         if inference:
             x = self.encoder(x)
-            x = x.unsqueeze(0)
+            x = x.unsqueeze(0)  # for rnn input, put the seq_len of x as 1 instead of none.
+            # prev_state: DataType: {list{tuple{Tensor}}}; Initially, it is a list of None
```
Comment (PaParaZz1) [inline ding/model/template/q_learning.py:686]: using python typing mask: `List[Tuple[torch.Tensor]]`

File: ding/model/template/q_learning.py [hunk 2]
```diff
@@ -700,11 +703,14 @@ def forward(
                     saved_hidden_state.append(prev_state)
                 lstm_embedding.append(output)
                 hidden_state = list(zip(*prev_state))  # {list: 2{tuple: B{Tensor:(1, 1, head_hidden_size}}}
+                # only keep ht, {list: x.shape[0]{Tensor:(1, batch_size, head_hidden_size)}}
                 hidden_state_list.append(torch.cat(hidden_state[0], dim=1))
             x = torch.cat(lstm_embedding, 0)  # (T, B, head_hidden_size)
             x = parallel_wrapper(self.head)(x)  # (T, B, action_shape)
-            x['next_state'] = prev_state  # the last timestep state including h and c
-            x['hidden_state'] = torch.cat(hidden_state_list, dim=-3)  # the all hidden state h
+            # the last timestep state including h and c, {list: B{tuple: 2{Tensor:(1, 1, head_hidden_size}}}
```
Comment (PaParaZz1) [inline ding/model/template/q_learning.py:710]: add `for lstm`

File: ding/policy/r2d2.py [hunk 3]
```diff
@@ -332,7 +339,7 @@ def _forward_learn(self, data: dict) -> Dict[str, Any]:
         return {
             'cur_lr': self._optimizer.defaults['lr'],
             'total_loss': loss.item(),
-            'priority': td_error_per_sample.abs().tolist(),
+            'priority': td_error_per_sample.abs().tolist(),  # TODO no need to add abs actually
```
Comment (PaParaZz1) [inline ding/policy/r2d2.py:342]: why no need to abs

**Review by PaParaZz1 — APPROVED** [review]
